src/coreclr/jit/compiler.h

```diff
@@ -10967,8 +10967,10 @@ class Compiler
     unsigned typGetBlkLayoutNum(unsigned blockSize);
     // Get the layout for the specified array of known length
     ClassLayout* typGetArrayLayout(CORINFO_CLASS_HANDLE classHandle, unsigned length);
-    // Get the number of a layout for the specified array of known length
-    unsigned typGetArrayLayoutNum(CORINFO_CLASS_HANDLE classHandle, unsigned length);
+    // Get a layout like an existing layout, with all gc refs removed
+    ClassLayout* typGetNonGCLayout(ClassLayout* existingLayout);
+    // Get a layout like an existing layout, with all gc refs changed to byrefs
+    ClassLayout* typGetByrefLayout(ClassLayout* existingLayout);
```

Comment:
This feels like something that only stack allocation is ever going to use, so it would make more sense to me to keep it in objectalloc.h/cpp.